tools/pot/openvino/tools/pot/engines/ie_engine.py

```diff
@@ -233,7 +233,12 @@ def _fill_input(self, model, image_batch):
         if len(input_info) == 1:
             input_blob = next(iter(input_info))
             input_blob_name = self._get_input_any_name(input_blob)
-            return {input_blob_name: np.stack(image_batch, axis=0)}
+            image_batch = {input_blob_name: np.stack(image_batch, axis=0)}
+            if Shape(image_batch[input_blob_name].shape) != input_info[0].shape:
+                raise ValueError(f"Incompatible input shapes. "
+                                 f"Cannot infer {Shape(image_batch[input_blob_name].shape)} into {input_info[0].shape}."
+                                 "Try to specify the layout of the model.")
+            return image_batch
 
         if len(input_info) == 2:
```

Comment:
I think that we should check Shape's here too. This condition is actual for faster-/mask-rcnn models.